File: fast_sync_dart_client/fast_sync_client/example/lib/http/http_manager.dart
import 'dart:async';
import 'dart:convert';
import 'package:dio/dio.dart';
import 'package:fast_sync_client/fast_sync_client.dart';

class HttpManager implements IhttpManager {
  final dio = Dio();
  HttpManager();

  @override
  Future<SyncPayload> pull(SyncOperationMetadata metadata) async {
    Response response = await dio.post(
        'http://192.168.178.24:3000/express/pull',
        data: json.encode(metadata.toJson()));
    if (response.data["success"] == true) {
      return SyncPayload.fromJson(response.data["data"]);
    }
    return throw Exception('pull failed');
  }

  @override
  Future<bool> push(SyncPayload payload) async {
    Response response = await dio.post(
        'http://192.168.178.24:3000/express/push',
        data: json.encode(payload.toJson()));
    if (response.data["success"] == true) {
      return true;
    }
    return throw Exception('push failed');
  }

  @override
  Future<SyncPayload> sync(SyncOperationMetadata metadata) async {
    return SyncPayload();
  }
}

File: fast-sync-typescript-client-sdk/packages/exemple/src/http/http_manager.ts
import {
  SyncPayload,
  IHttpManager,
  SyncOperationMetadata,
} from "fast-sync-client";

import axios from "axios";

export class HttpManager implements IHttpManager {
  constructor() {}

  async pull(metadata: SyncOperationMetadata): Promise<SyncPayload> {
    try {
      const response = await axios.post(
        "http://localhost:3000/express/pull",
        metadata
      );
      if (response.data.success === true) {
        return SyncPayload.create(response.data.data);
      }
      throw new Error("pull failed");
    } catch (error) {
      throw new Error("pull failed");
    }
  }

  async push(payload: SyncPayload): Promise<boolean> {
    try {
      const response = await axios.post(
        "http://localhost:3000/express/pull",
        payload
      );
      if (response.data.success === true) {
        return true;
      }
      throw new Error("push failed");
    } catch (error) {
      throw new Error("push failed");
    }
  }
}
